### PR TITLE
Fix/order columns

### PR DESCRIPTION
**Pull request by ssean1**
- Column headers are now dynamically generated
- Column headers toggle sort ascending/descending when clicked
- Default column order no longer overrides request.GET.get('orderX'), is now hard-coded
- Comments added for clarity

### CODE REVIEW

**Comment by ssean1** [comment]
Found a bug -- when a column is sorted in one table and a user tries to go to the next page on the other table, the order for that table is not selected by default

File: td/projects/views.py [hunk 1]
```diff
@@ -61,9 +87,16 @@ def all(request):
         Q(percent_complete=100) &
         Q(status__contains="Complete") &
         Q(end_date__gte=timezone.now() + datetime.timedelta(-90))
-    ).order_by(order2)
+    ).order_by(order2 or '-end_date')  # Default sort for Recent Projects
+
     p2 = Paginator(recent_projects, ITEMS_PER_PAGE)
 
+    # See above comment on similar loop
+    if order2:
```
Comment (mdj2) [inline td/projects/views.py]: You have this same pattern of code twice. Could it be refactored? I'm thinking `if colheaders[item] in [order1, order2]:` might work.

File: td/projects/views.py [hunk 2]
```diff
@@ -50,9 +67,18 @@ def all(request):
         Q(status="Cancelled") |
         Q(status="No Status") |
         Q(status__contains="Completed")
-    ).order_by(order1)
+    ).order_by(order1 or 'account_name')  # Default sort for Active Projects
```
Comment (mdj2) [inline td/projects/views.py]: `order1` and `order2` both come from request.GET and aren't sanitized. Django will protect you from SQL injection, but if a user manually sets the GET parameter like `order1=random_column`, Django will throw a Server 500.

I believe the original code had the same issue. Maybe check to see if orderN exists in colheaders.values() before using it?

**Comment by ssean1** [comment]
@mdj2 The only known bug I've found is when you first load the page, if you try to sort the first table by 'account_name', it will reload the page and sort ascending, which is the same as the default value. Not sure if it's fix-able with one dictionary generating the colheaders

**Comment by mdj2** [comment]
I'm going to merge this with a couple changes. `list` is a python builtin function (which is frequently used) so it isn't a good idea to use that as a variable name (it will override the builtin function in that scope). Also, it helps to "self-document" your code if you name the variable something like "valid_order_names".

`[colheaders[item] for item in colheaders]` can be more Pythonically written as `list(colheaders.values())`

A dictionary's `.items()` method is almost always what you want to use in a `for` loop, since it makes it easier to access the key and the value.

Overall, nice job. The logic to do this correctly was trickier than I thought.

**Pull request merged**